### PR TITLE
Allow any datum to be pooled, including images

### PR DESCRIPTION
**Pull request by MrPerson**
Tested on a non-ss13 project and works so it should work here.

Renamed the file from atom_pool to pool

I'll try to recap any other changes here:
Changed pick_n_take to pop when grabbing stuff out of the pool.
Contents is excluded from vars set to initial (would runtime with images), but it gets reset manually by atom/movable.

### CODE REVIEW

File: code/game/pooling/pool.dm [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+		else
+			if(AM)
+				AM.loc = second_arg
+			pooled.New(second_arg)
+
+		return pooled
+
+
+/proc/PlaceInPool(var/datum/diver, destroy = 1)
```
Comment (RemieRichards) [inline code/game/pooling/pool.dm]: >argument renamed to diver
>pool
>diver

10/10

**Comment by optimumtact** [comment]
>Tested on a non-ss13 project and works so it should work here.
I'll have some of what you're taking

File: code/game/pooling/pool.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/game/pooling/pool.dm]: :+1:

**Comment by Razharas** [comment]
Do things really need resetting both on placing in list and on taking from it?

**Pull request merged**